index.js

```diff
@@ -34,14 +34,19 @@ var rebaseRelativePath = memoize(function(sourceFile, relativeRoot, relativePath
   // join relative path to root (e.g. 'src/' + 'file.js')
   var relativeRootedPath = relativeRoot ? path.join(relativeRoot, relativePath) : relativePath;
 
+  // Ultimately these will be used in URLs so normalise to URL path separators before comparison
+  relativeRootedPath = relativeRootedPath.replace(/\\/g, '/');
+  sourceFile = sourceFile.replace(/\\/g, '/');
+
   if (sourceFile === relativeRootedPath ||    // same path,
+      path.dirname(sourceFile) === path.dirname(relativeRootedPath) || // same dir, different file (e.g. foo.ts > foo.js)
       pathIsAbsolute(relativeRootedPath) ||   // absolute path, nor
```

Comment:
please fix the alignment of `||` here, they were all in the same column for improved readability.
Also cut the comment a bit shorter, (remove the `e.g.` part).